galsim/catalog.py

```diff
@@ -343,3 +349,255 @@ def __hash__(self): return hash(repr(self))
 
 
 
+class OutputCatalog(object):
+    """A class for building up a catalog for output, typically storing truth information
+    about a simulation.
+
+    Each row corresponds to a different object, and each column stores some item of
+    information about that object (e.g. flux or half_light_radius).
+
+    Note: no type checking is done when the data are added in add_row().  It is up to 
+    the user to make sure that the values added for each row are compatible with the
+    types given here in the `types` parameter.
+
+    Initialization
+    --------------
+
+    @param names    A list of names for the output columns.
+    @param types    A list of types for the output columns. [default: None, which assumes all
+                    columns are float]
+
+    Attributes
+    ----------
+
+    After construction, the following attributes are available:
+
+        nobjects    The number of objects so far in the catalog.
+        ncols       The number of columns in the catalog.
+        names       The names of the columns.
+        types       The types of the columns.
+        cols        The columns of data that have been accumulated so far.
+
+    """
+    # Watch out for this "Gotcha".  Using _rows=[] as the default argument doesn't work!
+    # https://pythonconquerstheuniverse.wordpress.com/2012/02/15/mutable-default-arguments/
```

Comment:
FWIW, my preferred workaround for this is use `_rows=()` as the argument and default value, then `self.rows = list(_rows)` to set the attribute.  But whether that's better or not depends on how much you trust the user to pass a list rather than some other sequence, and given the underscore in the name, I'm guessing that's not a concern here.